lib/AppInfo/Application.php

```diff
@@ -39,6 +39,33 @@ class Application extends App implements IBootstrap {
 	public const DEFAULT_TRANSCRIPTION_MODEL_ID = 'whisper-1';
 	public const DEFAULT_IMAGE_SIZE = '1024x1024';
 	public const MAX_GENERATION_IDLE_TIME = 60 * 60 * 24 * 10;
+	public const DEFAULT_MAX_NUM_OF_TOKENS = 1000;
+	public const DEFAULT_QUOTA_PERIOD = 30;
+	
+	public const QUOTA_TYPE_TEXT = 0;
+	public const QUOTA_TYPE_IMAGE = 1;
+	public const QUOTA_TYPE_TRANSCRIPTION = 2;
+
+	
+	public const QUOTA_TYPES = [
+		self::QUOTA_TYPE_TEXT => 'text',
+		self::QUOTA_TYPE_IMAGE => 'image',
+		self::QUOTA_TYPE_TRANSCRIPTION => 'transcription',
+	];
+	public const DEFAULT_QUOTAS = [
+		self::QUOTA_TYPE_TEXT => ['type' => self::QUOTA_TYPES[self::QUOTA_TYPE_TEXT], 'value' => 0], // 0 = unlimited
```

Comment:
Is there a reason that the value of the array key has a type key with the value of the array key again? Couldn't we do 

```suggestion
		self::QUOTA_TYPE_TEXT =>  0, // 0 = unlimited
```

instead?

Reply:
Type is not actually equal to the array key, which is numeric but the text representation.  Here I semi-transitioned to using associative arrays which is kinda funny as I didn't go all the way. The point was to have a text representation on the js side. In my C oriented mind I was avoiding having a separate enum (that would need to match the "enum" here) on the js side so as to keep the code understandable.

Reply:
Fixed in https://github.com/nextcloud/integration_openai/pull/54/commits/44720e0aebfd3f549a6d9bd48a78df2872f26110